lms/djangoapps/bulk_email/views.py

```diff
@@ -0,0 +1,64 @@
+"""
+Views to support bulk email functionalities like opt-out.
+"""
+
+from __future__ import division
+
+import logging
+
+from six import text_type
+
+from django.contrib.auth.models import User
+from django.http import Http404
+from django.views.decorators.http import require_GET
+
+from bulk_email.models import Optout
+from courseware.courses import get_course_by_id
+from edxmako.shortcuts import render_to_response
+from notification_prefs.views import (
+    UsernameCipher,
+    UsernameDecryptionException,
+)
+
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+
+
+log = logging.getLogger(__name__)
+
+
+@require_GET
+def opt_out_email_updates(request, token, course_id):   # pylint: disable=unused-argument
+    """
+    A view that let users opt out of any email updates.
+
+    This meant is meant to be the target of an opt-out link or button.
+    The request must be a GET, and the `token` parameter must decrypt
+    to a valid username. The `course_id` is the course id of any course.
+
+    A 405 will be returned if the request method is not GET. A 404 will be
+    returned if the token parameter does not decrypt to a valid username.
+    It returns status 204 (no content) or an error.
+    """
+    try:
+        username = UsernameCipher().decrypt(token.encode())
+        user = User.objects.get(username=username)
+        course_key = CourseKey.from_string(course_id)
+        course = get_course_by_id(course_key, depth=0)
+    except UnicodeDecodeError:
+        raise Http404("base64url")
+    except UsernameDecryptionException as exn:
+        raise Http404(text_type(exn))
+    except User.DoesNotExist:
+        raise Http404("username")
+    except InvalidKeyError:
+        raise Http404("course")
+
+    Optout.objects.get_or_create(user=user, course_id=course_key)
```

Comment:
As it is changing data on the server, I would make it a POST request rather than a GET.

Reply:
@awaisdar001 Yep, good point.

Would it be ok to make the [unsubscribe page](https://github.com/edx/edx-platform/pull/19018/files#diff-34def588397cf14ad057eb362b58b834) be a confirmation form, so the user just says "yes" to submit the post request? That means I can keep the "unsubscribe" link in the email as a link, which is more text friendly.

Will work on this change in the next 2 weeks.